src/masoniteorm/connections/decorators.py

```diff
@@ -0,0 +1,20 @@
+from config.database import db
+
+
+class Transaction(object):
+
+    def __init__(self, connection="default"):
+        self.connection = connection
+
+    def __call__(self, fn, *args, **kwargs):
+        # before function
+        db.begin_transaction(self.connection)
+        try:
+            result = fn(*args, **kwargs)
```

Comment:
With that being said I think the solution is to just change this to:

```python
try:
    return fn
except:
    db.rollback(self.connection)
    return None
finally:
    db.commit(self.connection)
```

Reply:
notice we're not calling fn anymore

Reply:
Thank you I did not see this coming, should have paid more attention !